boilerplate/app/components/Toggle.tsx

```diff
@@ -0,0 +1,493 @@
+   * For switch, this affects the knob View.
+   */
+  inputDetailStyle?: StyleProp<ViewStyle | ImageStyle>
+  /**
+   * The position of the label relative to the action component.
+   * Default: right
+   */
+  labelPosition?: "left" | "right"
+  /**
+   * The label text to display if not using `labelTx`.
+   */
+  label?: TextProps["text"]
+  /**
+   * Label text which is looked up via i18n.
+   */
+  labelTx?: TextProps["tx"]
+  /**
+   * Optional label options to pass to i18n. Useful for interpolation
+   * as well as explicitly setting locale or translation fallbacks.
+   */
+  labelTxOptions?: TextProps["txOptions"]
+  /**
+   * Style overrides for label text.
+   */
+  labelStyle?: StyleProp<TextStyle>
+  /**
+   * Pass any additional props directly to the label Text component.
+   */
+  LabelTextProps?: TextProps
+  /**
+   * Special prop for the switch variant that adds a text/icon label for on/off states.
+   */
+  switchAccessibilityMode?: "text" | "icon"
+}
+
+interface ToggleInputProps {
+  on: boolean
+  status: ToggleProps["status"]
+  disabled: boolean
+  innerStyle: StyleProp<any>
+  detailStyle: StyleProp<any>
+  switchAccessibilityMode?: ToggleProps["switchAccessibilityMode"]
+}
+
+/**
+ * Renders a boolean input.
+ * This is a controlled component that requires an onValueChange callback that updates the value prop in order for the component to reflect user actions. If the value prop is not updated, the component will continue to render the supplied value prop instead of the expected result of any user actions.
+ *
+ * - [Documentation and Examples](https://github.com/infinitered/ignite/blob/master/docs/Components-Toggle.md)
+ */
+export function Toggle(props: ToggleProps) {
+  const {
+    variant = "checkbox",
+    editable = true,
+    status,
+    value,
+    onPress,
+    onValueChange,
+    labelPosition = "right",
+    inputOuterStyle: $inputOuterStyleOverride,
+    containerStyle: $containerStyleOverride,
+    ...WrapperProps
+  } = props
+
+  const disabled = editable === false || status === "disabled"
+
+  const Wrapper = useMemo<ComponentType<TouchableOpacityProps>>(
+    () => (disabled ? View : TouchableOpacity),
+    [disabled],
+  )
+  const ToggleInput = useMemo(() => ToggleInputs[variant] || (() => null), [variant])
+
+  const $containerStyle = [$container, $containerStyleOverride]
+
+  const $inputOuterStyle = [
+    $inputOuterVariants[variant],
+    value && { borderColor: colors.palette.secondary500 },
+    !value && { borderColor: colors.palette.neutral800 },
+    status === "error" && { borderColor: colors.error },
+    disabled && {
+      borderColor: colors.palette.neutral300,
+      backgroundColor: colors.palette.neutral300,
+    },
+    !disabled && { backgroundColor: colors.palette.neutral200 },
+    $inputOuterStyleOverride,
+  ]
+
+  function handlePress(e: GestureResponderEvent) {
+    if (disabled) return
+    onValueChange?.(!value)
+    onPress?.(e)
+  }
```

Comment:
Since this is a "controlled" component (same as RN's `Switch`) pressing on it won't change it's value. A handler needs to change the `value` prop in order for the component to change state. 2 convenience methods are provided - one with the touch event and the other with the potential new value. Either can be used to determine the `next` state of the Toggle.